lib/mock/mock-agent.js

```diff
@@ -134,6 +137,92 @@ class MockAgent extends Dispatcher {
   [kGetNetConnect] () {
     return this[kNetConnect]
   }
+
+  [kGetMockAgentClientsByConsumedStatus] () {
+    const mockAgentClients = this[kClients]
+
+    return Array.from(mockAgentClients.entries())
+      .map(([origin, scopeWeakRef]) => {
+        const scope = scopeWeakRef.deref()
+
+        if (scope == null) {
+          throw new Error('scope was null; this should not happen')
+        }
+
+        return { origin, scope }
+      })
+      .map(({ origin, scope }) => {
+        // @ts-expect-error TypeScript doesn't understand the symbol use
+        const clients = scope[kDispatches]
+
+        const consumed = clients.filter(({ consumed }) => consumed)
+        const unconsumed = clients.filter(({ consumed }) => !consumed)
+        const persistent = unconsumed.filter(({ persist }) => persist)
+        const tooFewUses = unconsumed.filter(({ persist }) => !persist)
+
+        return { origin, clients, consumed, persistent, tooFewUses }
+      })
+      .reduce(
+        (all, current) => ({
+          totals: {
+            consumed: all.totals.consumed.concat(current.consumed),
+            persistent: all.totals.persistent.concat(current.persistent),
+            tooFewUses: all.totals.tooFewUses.concat(current.tooFewUses)
+          },
+          clients: all.clients.concat({
+            origin: current.origin,
+            clients: current.clients
+          })
+        }),
+        {
+          totals: {
+            consumed: [],
+            persistent: [],
+            tooFewUses: []
+          },
+          clients: []
+        }
+      )
+  }
+
+  pendingInterceptors () {
+    return this[kGetMockAgentClientsByConsumedStatus]().totals.tooFewUses
+  }
+
+  assertNoUnusedInterceptors (options = {}) {
+    const clients = this[kGetMockAgentClientsByConsumedStatus]()
+
+    if (clients.totals.tooFewUses.length === 0) {
+      return clients
+    }
+
+    const interceptorPluralizer = new Pluralizer('interceptor', 'interceptors')
+    const tooFew = interceptorPluralizer.pluralize(
+      clients.totals.tooFewUses.length
+    )
+    const consumed = interceptorPluralizer.pluralize(
+      clients.totals.consumed.length
+    )
+    const persistent = interceptorPluralizer.pluralize(
+      clients.totals.persistent.length
+    )
+
+    throw new Error(`
```

Comment:
Could you use our error facility instead? https://github.com/nodejs/undici/blob/main/lib/core/errors.js